packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -743,6 +768,25 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     assert(() { _debugLocked = false; return true; });
   }
 
+  /// Returns the value of the current route's `willPop` method. This method is
+  /// typically called before a user-initiated [pop]. For example on Android it's
+  /// called by the binding for the system's back button.
+  ///
+  /// See also:
+  ///
+  /// * [Form], which provides an `onWillPop` callback that enables the form
+  ///   to veto a [pop] initiated by the app's back button.
+  /// * [ModalRoute], which provides a `scopedWillPopCallback` that can be used
+  ///   to define the route's `willPop` method.
+  Future<bool> willPop() async {
+    final Route<dynamic> route = _history.last;
+    assert(route._navigator == this);
+    if (!await route.willPop() && route.isCurrent)
```

Comment:
shouldn't the route check the isCurrent thing?
I think this can all be simplified to just `return route.willPop()`, if the route verifies isCurrent.